src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,105 @@
+package com.internet.shop.dao.jdbc;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import com.internet.shop.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+
+    @Override
+    public Product create(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO products(name, price) VALUES(?, ?)";
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, item.getName());
+            preparedStatement.setBigDecimal(2, item.getPrice());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong("product_id"));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new RuntimeException("Product " + item + " was not created", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long item) {
+        String exceptionMessage = "Can`t get product with id - " + item;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM products WHERE product_id = ? AND deleted != 1";
```

Comment:
```suggestion
            String query = "SELECT * FROM products WHERE product_id = ? AND deleted = 0";
```

Reply:
They are the same since the default value is 0.